File: 1582.special-positions-in-a-binary-matrix/solution.go
func numSpecial(mat [][]int) int {
    rows := make([]int, len(mat))
    cols := make([]int, len(mat[0]))
    for i := range mat {
        for j := range mat[0] {
            rows[i] += mat[i][j]
            cols[j] += mat[i][j]
        }
    }
    res := 0
    for i := range rows {
        if rows[i] != 1 {
            continue
        }
        for j := range cols {
            if mat[i][j] == 1 && cols[j] == 1 {
                res++
            }
        }
    }
    return res
}

File: 1582.special-positions-in-a-binary-matrix/solution.kt
class Solution {
    fun numSpecial(mat: Array<IntArray>): Int {
        val m = mat.size
        val n = mat[0].size
        val rs = IntArray(m)
        val cs = IntArray(n)
        
        for (i in 0 until m) {
            for (j in 0 until n) {
                rs[i] += mat[i][j]
                cs[j] += mat[i][j]
            }
        }

        var res = 0

        for (i in 0 until m) {
            if (rs[i] != 1) continue
            for (j in 0 until n) {
                if (cs[j] == 1 && mat[i][j] == 1) {
                    res++
                }
            }
        }

        return res
    }
}